lib/whenever/cron.rb

```diff
@@ -12,7 +12,25 @@ def initialize(time = nil, task = nil, at = nil)
         @at_given = at
         @time = time
         @task = task
+        
+        at = randomize_at(at,time)
+        
         @at   = at.is_a?(String) ? (Chronic.parse(at) || 0) : (at || 0)
+
+      end
+
+      def randomize_at(at, time)
+        return at unless at.nil?
+        unless time.is_a?(Symbol) && [:hour,:day].include?(time)
+          #puts 'Sorry, the randomizing-feature works only for symbols :hour and :day'
+          return at
+        end
+
+        case time
+          when :hour  then return Random.rand(0...59)
+          when :day   then return [23,0,1,2,3,4,5,6,7].sample.to_s + ':' + Random.rand(0...59).to_s
+        end
+        return at
```

Comment:
The `case` statement allows an `else` case, which would allow all of the "legs" of it to be the _last expression of the method_ = to return the value from the method = to drop the `return` keyword.